### ISSUE TITLE
net/mail: parses invalid address

### ISSUE THREAD

**Issue by dvyukov** [golang/go#11293]
The following program fails with the panic:
```go
package main

import (
	"fmt"
	"net/mail"
)

func main() {
	data := "\"\"@0"
	addr, err := mail.ParseAddress(data)
	if err != nil {
		return
	}
	_, err = mail.ParseAddress(addr.String())
	if err != nil {
		fmt.Printf("failed to parse addr: %q -> %q\n", data, addr)
		panic(err)
	}
}
```
```
failed to parse addr: "\"\"@0" -> "<@0>"
panic: mail: invalid string
```

I guess that address should have not been parsed in the first place.

go version devel +514014c Thu Jun 18 15:54:35 2015 +0200 linux/amd64

**Comment by dvyukov** [comment]
This is probably the same issue:
```
	data := "\"0:\"@0"
```
but worth a separate test.

**Comment by dvyukov** [comment]
One more:
```
    data := "\"\t0\"@0"
```

**Comment by DenBeke** [comment]
What is wrong with `data := "\"0:\"@0"` ?

**Comment by dvyukov** [comment]
It does not parse successfully in the second mail.ParseAddress call. Either first mail.ParseAddress or String corrupt it somehow, or it should not be parsed the first time as well.

**Comment by DenBeke** [comment]
I recently did a patchrequest to fix that round trip issue #10768: https://github.com/golang/go/commit/daaa45073e887cb6d8075c2caafdfe8425bca25a (will be included in Go 1.5 afaik)

First and third example should indeed fail to parse, which doesn't happen. If I find some time I can look into it, doesn't seem hard to solve.

**Comment by DenBeke** [comment]
I uploaded a patch request for this issue...  
https://golang.org/cl/12905
